docs/data/data-grid/export/export.md

```diff
@@ -185,12 +185,69 @@ The demo below shows how to add a JSON export.
 
 {{"demo": "CustomExport.js", "bg": "inline", "defaultCodeOpen": false}}
 
-## 🚧 Excel export [<span class="plan-premium"></span>](https://mui.com/store/items/material-ui-pro/)
+## Excel export [<span class="plan-premium"></span>](https://mui.com/store/items/material-ui-pro/)
 
-> ⚠️ This feature isn't implemented yet. It's coming.
+> ⚠️ This feature is temporarily available on the Pro plan until the release of the Premium plan.
+>
+> To avoid future regression for users of the Pro plan, the feature needs to be explicitly activated using the `excelExport` experimental feature flag.
 >
-> 👍 Upvote [issue #198](https://github.com/mui/material-ui-x/issues/198) if you want to see it land faster.
-> You will be able to export the displayed data to Excel with an API call, or using the grid UI.
+> ```tsx
+> <DataGridPro experimentalFeatures={{ excelExport: true }} {...otherProps} />
+> ```
+>
+> The feature is stable in its current form, and we encourage users willing to migrate to the Premium plan once available to start using it.
+
+The Excel export allows translating columns' type and tree structure of a DataGrid to an Excel file.
+
+Columns with type `'boolean'`, `'number'`, `'singleSelect'`, `'date'`, and `'dateTime'` will be exported in their corresponding type in Excel. To make this work, please ensure that `rows` values have the correct type, or [convert types to correct ones](/components/data-grid/columns/#converting-types).
+
+{{"demo": "ExcelExport.js", "bg": "inline", "defaultCodeOpen": false}}
+
+This feature relies on the package [exceljs](https://github.com/exceljs/exceljs).
+To export in Excel format, you have to install this package.
+
+```sh
+ // with npm
+ npm install exceljs
+
+ // with yarn
+ yarn add exceljs
+```
+
+### Template Excel
+
+You can customize the document before and after the creation of rows containing data, by using functions `exceljsPreProcess` and `exceljsPostProcess`.
```

Comment:
I do not think we can change the lib without making breaking change.

If we change the library, it will have different methods, and so developers will have to rewrite there `excelPreProcessor` and  `excelPostProcessor`

To add a new library, we can add an option `library` set by default to `"exceljs"` and if it developpers set it to `"xlsx"` we use that library and `xlsxPreProcessor` instead of `exceljsPreProcessor`